src/consensus/replay_tower.zig

```diff
@@ -1374,10 +1367,24 @@ pub const ReplayTower = struct {
             progress,
             epoch_stake.total_stake,
             &epoch_stake.stakes.vote_accounts.vote_accounts,
-            latest_validator_votes_for_frozen_banks,
+            latest_validator_votes,
             fork_choice,
         );
 
+        const maybe_slot_history: ?SlotHistory =
+            if (heaviest_slot_on_same_voted_fork) |heaviest| blk: {
+                var ancestors_for_heaviest = ancestors.get(heaviest) orelse
+                    return error.MissingAncestors;
+                var slot_ancestors = sig.core.Ancestors.EMPTY;
+                for (ancestors_for_heaviest.items()) |ancestor_slot| {
+                    try slot_ancestors.addSlot(allocator, ancestor_slot);
+                }
+                break :blk try slot_history_accessor.getSlotHistory(allocator, &slot_ancestors);
```

Comment:
It's unfortunate that you need to allocate to clone the ancestors only to immediately discard it. Also, this leaks.

Ideally you could just use a pointer to the existing data. I see two options:
1. shallow-copy `ancestors_for_heaviest` into Ancestors - this has ambiguous ownership semantics, but since it's only passed by const pointer and then immediately discarded, it's not a big problem.
2. change the ancestors map at the source to be a map from Slot to Ancestors instead of a SlotSet, and get it with `getPtr`

My instinct would be 2 unless that's impractical for some reason, then maybe I would go with 1. 1 is kind of a hack, so I would leave a comment there about the ownership and lifetimes if you take that approach.

Reply:
Went with 2 as that feels more correct. Although it ended up being more intrusive [5c6268d](https://github.com/Syndica/sig/pull/810/commits/5c6268d3fe7b3d9d15ca39614b736cd436dd32ae)
[8da86ea](https://github.com/Syndica/sig/pull/810/commits/8da86ea2b708c4677405703cfc1abb61d0cb988d)